kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/SchemaUtils.java

```diff
@@ -79,6 +82,118 @@ public static void validateSchema(StructType schema, boolean isColumnMappingEnab
     validateSupportedType(schema);
   }
 
+  /**
+   * Performs the following validations on an updated table schema:
+   *
+   * <p>The following checks are performed:
+   *
+   * <ul>
+   *   <li>No duplicate columns are allowed
+   *   <li>Column names contain only valid characters
+   *   <li>Data types are supported
+   *   <li>No new non-nullable fields are added
+   *   <li>Physical column name consistency is preserved in the new schema
+   *   <li>Partition columns are not modified
+   *   <li>Nested IDs for array/map types are preserved in the new schema
+   *   <li>No type changes
+   * </ul>
+   */
+  public static void validateUpdatedSchema(
+      StructType currentSchema,
+      StructType newSchema,
+      List<String> partitionCols,
+      Metadata metadata) {
+    validateSchema(newSchema, true /*columnMappingEnabled*/);
```

Comment:
since the `metadata` is an input here let's validate our assumption here that column mapping is enabled?

`checkArgument(metdata.columnMappingEnabled) // shorthand` or something?

Reply:
Good idea, done!